assessments/projects/openfga/self-assessment.md

```diff
@@ -0,0 +1,277 @@
+
+### Actions
+
+**Invoking the OpenFGA APIs**
+
+Every time a server endpoint is invoked, OpenFGA validates that:
+
+  - The credentials provided in the API call match the ones configured in the server.
+
+  - Ensures that the input is a semantically valid, e.g. that a tuple is valid according to the authorization model or that the model does not have disallowed cyclical or problematic definitions
+
+  - Payload Verification: 
+  
+    - Confirm that API payloads adhere to Protobuf API definitions.  
+    - Validate parameters for proper structure, e.g. ensuring users are written in the correct format which is 'user:<userid>'
+  
+**Writing an Authorization Model**
+
+  - Semantic Model Verification: OpenFGA validates that Authorization Models are semantically valid, avoiding cyclical or problematic definitions and other disallowed criteria.
+
+  - Version and Configuration: When writing a model, a new version is created (models are immutable). Applications must be configured to use the new version after validation and confirmation of expected behavior.
+
+**Calling the Authorization Query endpoints**
+
+When the [/check](https://openfga.dev/api/service#/Relationship%20Queries/Check) and [/list-objects](https://openfga.dev/api/service#/Relationship%20Queries/ListObjects) endpoints are called, OpenFGA limits the number of simultaneous paths explored and enforces depth limitations on the graph traversal. 
+
+To protect against DoS attacks, OpenFGA restricts both the number of simultaneous paths explored and the depth of paths traversed in the graph.
+
+**Upgrading OpenFGA Database Schema**
+
+Database Migration Planning: When installing new versions of OpenFGA, database migrations can be executed in a way that minimizes downtime and ensure a smooth transition in the target system.
+
+### Goals
+
+* Simplify and standardize authorization processes, making them more consistent across various applications and systems.
+
+* Establish patterns and standards for externalized authorization.
+
+* Create architectural patterns, terminologies, and protocols that enable interoperability among different authorization systems.
+
+* Deliver an authorization service for any application component.
+
+* Enable centralized authorization decisions and permits diverse teams to implement authorization using a shared framework across various application components.
+
+### Non-Goals
+
+* Tools for management of groups/roles/permissions not inherently provided to the end-users.
+* Does not intend to serve as a comprehensive data repository for non-authorization related data.
+* Does not aim to provide a complete authentication and Access Control Solution.
+
+## Self-Assessment Use
+
+This self-assessment is created by the OpenFGA team to perform an internal analysis of the project's security. It is not intended to provide a security audit of OpenFGA, or function as an independent assessment or attestation of OpenFGA's security health.

```

Comment:
I'm putting my comments for future reference when the team is considering syncing with OpenFGA security self-assessment. If can specific who in the OpenFGA team is driving this topic along with the security pals.